src/main/scala/mesosphere/marathon/api/TaskKiller.scala

```diff
@@ -34,32 +36,40 @@ class TaskKiller @Inject() (
       case Some(app) =>
         checkAuthorization(UpdateRunSpec, app)
 
+        // TODO: We probably want to pass the execution context as an implcit.
         import scala.concurrent.ExecutionContext.Implicits.global
-        taskTracker.appTasks(appId).flatMap { allTasks =>
+
+        async {
+          val allTasks = await(taskTracker.appTasks(appId))
           val foundTasks = findToKill(allTasks)
-          val expungeTasks = if (wipe) expunge(foundTasks) else Future.successful(())
-
-          expungeTasks.map { _ =>
-            val launchedTasks = foundTasks.filter(_.launched.isDefined)
-            if (launchedTasks.nonEmpty) {
-              service.killTasks(appId, launchedTasks)
-              foundTasks
-            } else foundTasks
-          }
+
+          if (wipe) expunge(foundTasks)
+
+          val launchedTasks = foundTasks.filter(_.launched.isDefined)
+          if (launchedTasks.nonEmpty) await(service.killTasks(appId, launchedTasks))
+          // TODO: For some reasons tasks returned by service.killTasks !=
+          // foundTasks in some cases. What should be returned?
+          foundTasks
         }
 
       case None => Future.failed(UnknownAppException(appId))
     }
   }
 
-  private[this] def expunge(tasks: Iterable[Task])(implicit ec: ExecutionContext): Future[Unit] = {
-    tasks.foldLeft(Future.successful(())) { (resultSoFar, nextTask) =>
-      resultSoFar.flatMap { _ =>
+  private[this] def expunge(tasks: Iterable[Task])(implicit ec: ExecutionContext): Unit = {
+    tasks.foreach { nextTask =>
+      async {
```

Comment:
I want to make it explicit that the tasks are not processed in parallel but rather sequentially.